functions/check-started/index.js

```diff
@@ -32,9 +32,11 @@ module.exports = async function (context, checkStartMessage) {
     throw error
   }
 
-  // Delete the row in the preparedCheck table - prevent pupils logging in again.
+  // Delete the row in the preparedCheck table for live checks only - prevent pupils logging in again.
   try {

```

Comment:
the problem with this is that you are trusting the client payload to be correct.  You should be looking this up in the main database via checkCode to determine how the flag is set.